webapp-client/src/errorify.js

```diff
@@ -0,0 +1,30 @@
+import ErrorMessage from './component/ErrorMessage';
+import React from 'react';
+
+/**
+ * allowErrorMessage modifies the given component to give it a boolean state
+ * variable called hasError. Along with this, the methods showErrorMessage,
+ * rendorErrorMessage and clearErrorMessage are added. These allow the component
+ * to easily make use of the ErrorMessage component.
+ * @param {React.Component} component - A react component
+ */
+export default function allowErrorMessage(component) {
+  component.setState({ hasError: false });
+  component.clearErrorMessage = function () {
+    component.setState({
+      hasError: false,
+    });
+  };
+  component.showErrorMessage = function () {
+    component.setState({ hasError: true });
+  };
+  component.renderErrorMessage = function (message) {
+    return (
+      <ErrorMessage
+        hasError={component.state.hasError}
+        message={message}
+        closePopup={component.clearHasError}
+      />
+    );
+  };
+}
```

Comment:
just curious, is there a reason why this file is in the src folder instead of the component folder?

Reply:
I actually don't know where it should go! For me, it was a semantic thing because it's not technically a component. Where would you propose that it could go?

Reply:
Ah that's valid! the first place i looked for it was the component folder but I see your point about it not being a component:))